### PR TITLE
feat(bridge-api): switch from hashed tokens to encrypted tokens

### LINKED ISSUE #84: Support for crypted tokens
Hashed tokens are deprecated.
See [https://developer.nuki.io/t/bridge-beta-fw-1-22-1-2-14-0-with-new-encrypted-bridge-http-api-token/15816](https://developer.nuki.io/t/bridge-beta-fw-1-22-1-2-14-0-with-new-encrypted-bridge-http-api-token/15816)

### PR DESCRIPTION
**Pull request by pree**
fixes #84

### CODE REVIEW

**Comment by pschmitt** [comment]
Thanks for implementing this! I wonder though if this works with older bridge (software) versions, I'm assuming it doesn't - right?

If that's the case we should make this pynuki 2.0 release, rather than a simple patch bump.

**Comment by pree** [comment]
[As for the docs](https://developer.nuki.io/page/nuki-bridge-http-api-1-13/4/#heading--token) it seems to be compatible even with older bridges:
```
Bridge 1.0: ≥1.22.1
Bridge 2.0: ≥2.14.0
```
That's why I decided to replace the hash_token instead of adding to it :) 
In their [release notes](https://developer.nuki.io/t/nuki-bridge-firmware-release-notes-public-releases/2797) for the public releases it seems to not be released yet (should have been in july), but maybe they just forgot to post it. As my bridge is in the beta channel I cannot confirm the current release.

I guess the only bridge that won't be supported anymore by pynuki would be the software bridge, which isn't supported by nuki anymore since nearly two years.

**Comment by pree** [comment]
If you don't mind, I'd love if you can add the `hacktoberfest-accepted' label to this PR :)

File: pynuki/utils.py [hunk 1]
```diff
@@ -5,15 +5,20 @@
 from datetime import datetime
 from random import randint
 
+import nacl.secret
```
Comment (pschmitt) [inline pynuki/utils.py:8]: The pynacl library should be added as a dependency in pyproject.toml.

**Comment by pschmitt** [comment]
This seems to break the discovery feature:

```
python -c "from pynuki import NukiBridge; NukiBridge.discover()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/pschmitt/devel/private/pynuki.git/pynuki/bridge.py", line 62, in discover
    return [
  File "/home/pschmitt/devel/private/pynuki.git/pynuki/bridge.py", line 63, in <listcomp>
    NukiBridge(hostname=x.get("ip"), port=x.get("port"))
  File "/home/pschmitt/devel/private/pynuki.git/pynuki/bridge.py", line 45, in __init__
    self.token = token
  File "/home/pschmitt/devel/private/pynuki.git/pynuki/bridge.py", line 78, in token
    self.__tokendigest = sha256sum(token)
  File "/home/pschmitt/devel/private/pynuki.git/pynuki/utils.py", line 15, in sha256sum
    return hashlib.sha256(text.encode("utf-8")).digest()
AttributeError: 'NoneType' object has no attribute 'encode'
```

**Comment by pree** [comment]
Thanks for the review and the fixes! :)

**Pull request merged**